### ISSUE TITLE
Improve error message for invalid #r reference

### ISSUE THREAD

**Issue by paulbatum** [Azure/azure-webjobs-sdk-script#185]
When using NuGet packages, you do not actually need to reference them with a #r and in fact, attempting to do so will result in a rather unintuitive error:

`#r "Microsoft.ProjectOxford.Vision"`

```
2016-03-28T23:07:56.997 Script for function 'ocrtest' changed. Reloading.
2016-03-28T23:07:56.997 Compiling function script.
2016-03-28T23:07:57.372 (5,1): error CS0006: Metadata file 'Microsoft.ProjectOxford.Vision' could not be found
```

The runtime should detect that the user is attempting to #r a NuGet and provide an understandable error message.

**Comment by fabiocav** [comment]
@paulbatum , is this verbiage clear?
```
error CS0006: Metadata file 'Microsoft.ProjectOxford.Vision' could not be found.
warning AF00x: The assembly 'Microsoft.ProjectOxford.Vision' is part of a referenced NuGet package. Package assemblies are automatically referenced by your Function and do not require a "#r" directive.
```

**Comment by paulbatum** [comment]
Yep looks good!

**Comment by fabiocav** [comment]
Change merged in, with a slight change to the message, which now looks like this:
```
The reference 'Foo.Bar' is part of the referenced NuGet package 'FooPackage'. Package assemblies are automatically referenced by your Function and do not require a '#r' directive.
```

**Comment by fabiocav** [comment]
Resolved in #227
